### PR TITLE
 Use defaults for images in operator

### LINKED ISSUE #284: Images in the CRD should be optional, with defaults handled by operator
We would like to have defaults for the images in the Observatorium CR. There are currently are 11 images to configure (6x Thanos,api,queryCache, memcached, memcached_exporter, receivecontroller). This could be a hurdle for the user.

1. Make images as optional in the CRD.
2. Operator will use preconfigured images if the user did not specified them in the CR.
3. The user can override one or more images.

### PR DESCRIPTION
**Pull request by rollandf**
Fixes #284

### CODE REVIEW

**Comment by rollandf** [comment]
Should we have an additional CR example to test this in CI?

File: operator/api/v1alpha1/observatorium_types.go [hunk 1]
```diff
@@ -55,14 +55,14 @@ type ObjectStorageConfig struct {
 
 type ThanosReceiveControllerSpec struct {
 	// Receive Controller image
-	Image string `json:"image"`
+	Image string `json:"image,omitempty"`
```
Comment (brancz) [inline operator/api/v1alpha1/observatorium_types.go:58]: I would argue that _if_ `image` is set, we want to force users to also set version. So they understand that they need to appropriately set `version` according to what's in the `image` they specify.

**Comment by rollandf** [comment]
> Love the direction! I'm trying to think of how this would work with productization, typically the red hat pipelines need it to be flags on the operator binary etc. and they are replaced by the machinery. Do we know how this would work? (I'm not requesting to solve this today, just we should think about it, we might just set the "defaults" via flags eventually)

I was thinking of having a downstream version of `default-config.libsonnet` that will have the downstream images fitting a specific version. Then is the Downstream Docker file just copy it instead of the upstream one.

File: operator/api/v1alpha1/observatorium_types.go (hunk 1 above)
Comment (rollandf) [reply operator/api/v1alpha1/observatorium_types.go:58]: Well, this is not especially related to this patch, but I had always doubts about the `version` field and for is the purpose of it.
I am not sure how to address your comment actually.

Comment (brancz) [reply operator/api/v1alpha1/observatorium_types.go:58]: I'd be curious if we could have conditional validation in the openapischema? Not blocking this I think, but we should have an issue to track this.

**Comment by brancz** [comment]
> I was thinking of having a downstream version of default-config.libsonnet that will have the downstream images fitting a specific version. Then is the Downstream Docker file just copy it instead of the upstream one.

At least looking at how things worked in OpenShift, this would not have worked, but let's figure out how to solve that when we're there :)

File: operator/api/v1alpha1/observatorium_types.go (hunk 1 above)
Comment (rollandf) [reply operator/api/v1alpha1/observatorium_types.go:58]: I open issue #288 for follow up

**Pull request merged**